master/internal/api_runs.go

```diff
@@ -239,3 +251,201 @@ func sortRuns(sortString *string, runQuery *bun.SelectQuery) error {
+					})
+				} else {
+					results = append(results, &apiv1.RunActionResult{
+						Error: "",
+						Id:    check.ID,
+					})
+				}
+				continue
+			}
+			if check.ExpID != nil {
+				expMoveIds = append(expMoveIds, *check.ExpID)
+			}
+			validIDs = append(validIDs, check.ID)
+		}
+	}
+	if req.Filter == nil {
+		for _, originalID := range req.RunIds {
+			if !slices.Contains(visibleIDs, originalID) {
+				results = append(results, &apiv1.RunActionResult{
+					Error: fmt.Sprintf("Run with id '%d' not found in project with id '%d'", originalID, req.SourceProjectId),
+					Id:    originalID,
+				})
+			}
+		}
+	}
+	if len(validIDs) > 0 {
+		tx, err := db.Bun().BeginTx(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+		defer func() {
+			txErr := tx.Rollback()
+			if txErr != nil && txErr != sql.ErrTxDone {
+				logrus.WithError(txErr).Error("error rolling back transaction in MoveRuns")
+			}
+		}()
+
+		expMoveResults, err := experiment.MoveExperiments(ctx, expMoveIds, nil, req.DestinationProjectId)
+		if err != nil {
+			return nil, err
+		}
+		failedExpMoveIds := []int32{-1}
+		for _, res := range expMoveResults {
+			if res.Error != nil {
+				failedExpMoveIds = append(failedExpMoveIds, res.ID)
+			}
+		}
+		var acceptedIDs []int32
+		if _, err = tx.NewUpdate().Table("runs").
+			Set("project_id = ?", req.DestinationProjectId).
+			Where("runs.id IN (?)", bun.In(validIDs)).
+			Where("runs.experiment_id NOT IN (?)", bun.In(failedExpMoveIds)).
+			Returning("runs.id").
+			Model(&acceptedIDs).
+			Exec(ctx); err != nil {
+			return nil, fmt.Errorf("updating run's project IDs: %w", err)
+		}
+
+		for _, acceptID := range acceptedIDs {
+			results = append(results, &apiv1.RunActionResult{
+				Error: "",
+				Id:    acceptID,
+			})
+		}
+		if err = tx.Commit(); err != nil {
+			return nil, err
+		}
+	}
+	return &apiv1.MoveRunsResponse{Results: results}, nil
+}
+
+func cloneExperimentAndRun(ctx context.Context, expID int32, runID int32, destProjID int32) error {
+	var cloneExpID int32
+
+	// clone experiment into dest project
+	err := db.Bun().NewRaw(`INSERT INTO experiments(state, config, model_definition, start_time, end_time, archived,
+		parent_id, owner_id, progress, original_config, notes, job_id, project_id, checkpoint_size, checkpoint_count,
+		best_trial_id, unmanaged, external_experiment_id) 
+		SELECT 'COMPLETED'::experiment_state as state, config, model_definition, start_time,
+		(CASE WHEN end_time IS NULL THEN NOW() ELSE end_time END) as end_time, archived, parent_id, owner_id, progress,
+		original_config, notes, job_id, ? as project_id, checkpoint_size, checkpoint_count, best_trial_id, unmanaged,
+		external_experiment_id FROM experiments WHERE id = ? RETURNING id;`, destProjID, expID).Scan(ctx, &cloneExpID)
+	if err != nil {
+		return err
+	}
+
+	var cloneRunID int32
+	// clone run into dest project
+	err = db.Bun().NewRaw(`INSERT INTO runs(experiment_id, state, start_time, end_time, hparams, warm_start_checkpoint_id,
```

Comment:
blocking: this clone doesn't include metrics or checkpoints or tasks right?

Does anything in the ui work? I wouldn't expect logs or charts to be populated on this cloned run

Reply:
This just copies the row that exists in the run table with a new id. This simply allows a user to view the run in a different project in the Run Table. The original run and associated metrics/checkpoints/tasks are unchanged and no new ones are made

Reply:
I haven't used the runs table but I'm assuming there is going to be a link to the runs detail page or the experiment page?

This experiment / run detail page won't be complete because of the shallow copy?

Reply:
After discussion, we have decided to pause on implementing the clone feature. For now we can either skip moving the multi-trial or move the entire experiment.